docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,107 @@
+# Storage Node Downtime Disqualification
+
+## Abstract
+
+This document describes storage node downtime suspension and disqualification.
+
+## Background
+
+The previous implementation of downtime disqualification led to the unfair disqualification of storage nodes. This prompted a halt to downtime disqualification and a new design for [Downtime Tracking](storage-node-downtime-tracking.md).
+
+While the previously mentioned blueprint presents a means of tracking downtime, it leaves out the design of how to use this information for disqualification. That is the purpose of this blueprint.
+
+## Design
+
+### Suspension
+
+Storage node downtime can have a range of causes. For those storage node operators who may have fallen victim to a temporary issue, we want to give them a chance to diagnose and fix it before disqualifying them for good. For this reason, we are introducing downtime suspension as a component of disqualification.
+
+For downtime suspension and disqualification we need a few new configurable values:
+
+- Tracking Period: The length of time into the past for which we measure downtime.
+- Allowed Downtime: The amount of downtime within the tracking period we will allow before suspension.
+- Grace Period: The length of time to give suspended SNOs to diagnose and fix issues causing downtime. Afterwards, they will have one tracking period to demonstrate an acceptable amount of downtime before disqualification.
+
+When the [downtime tracking system](storage-node-downtime-tracking.md) adds an entry to the nodes_offline_time table, if the node is not already suspended, we check the total amount of downtime for the tracking period, e.g. if the tracking period is 30 days, we sum downtime for the last 30 days. If the total is greater than the allowed downtime, the node is suspended for a length of time equal to the grace period + one tracking period.
+
+Suspended nodes are not selected for uploads, and all pieces they hold are considered unhealthy. 
+
+### Disqualifying or reinstating suspended nodes
+
+In order to be reinstated, suspended nodes need to prove to us that they have changed their behavior. This requires that we observe a period of acceptable downtime from them. Otherwise, they are disqualified.
+
+For this purpose we implement a new chore for evaluating suspended nodes. The evaluation chore runs on a configurable interval, and checks to see if any downtime-suspended nodes have reached the end of their suspension (grace period + tracking period). For these nodes, check the total downtime for the tracking period. If a node's downtime has fallen within the allowed downtime range, it is reinstated. Otherwise, it is disqualified.
```

Comment:
This will fail in the following example:
1. Storage node was offline 30 days ago but it was within the acceptable downtime. Storage node was not suspended.
2. Today storage node is offline for 10 minutes on top of the the downtime 30 days ago. Now it hits the magic line and gets suspended.
3. Expected behavior: A few minutes later the "check suspended chore" should notice that the downtime from 30 days ago is now out of scope. Total downtime 10 minutes. Node should get reinstated. Based on this design document the chore will wait the grace periode before evaluating. -> I would call that a bug.

Reply:
That's a good point. The node has proven that its behavior has changed. As you mention below, if we're okay to risk nodes going in and out of suspension by hovering near the suspension threshold, then I think we can evaluate continuously rather than wait a set amount of time.